fe/example/src/api/partnerWebhook.ts

```diff
@@ -0,0 +1,26 @@
+import { BE_BASE_URL } from './constants';
+
+const webhookUrl = (endpoint: string) => `${BE_BASE_URL}/webhook/${endpoint}`;
+
+/**
+ * You wouldn't have this in a real production system, but it's an easy way to
+ * seed data for demonstrative purposes without having to deploy a database,
+ * public webhook endpoint, etc.
+ */
+export const postPartnerWebhook = async (data: unknown, endpoint: string) => {
```

Comment:
:tada:

This is really good. We definitely need more ways to demonstrate webhooks before going fully prod.

I keep toying around with the idea of adding a mutation that would send a `CandidateApplicationCreated` for a given application ID to once-off URL. I suspect that would be a lot of work if we wanted to share code paths with real webhooks, though.

Reply:
Hmm, actually would this work:

1. Create a new e.g. `sendTestCandidateApplicationCreatedEvent` that takes a candidate OID and URL. Presumably there would be other `sendTest`... mutations later. This would duplicate the code to create the event body. It would probably need to go through our candidate application service's public test instance to validate and lookup the IDs.

2. Add a new `submitTestEvent` endpoint in our webhook service that takes a URL instead of a hirer ID. It would run all the normal security validation on the URL as if you were creating a new subscription.

3. Have our webhook service invoke our HTTP caller Lambda synchronously (skip persisting anything to the database). We could set a concurrency limit of 1 to avoid people spamming/DOSing through out caller (even though this would be Playground authenticated)

4. Return a pseudo-delivery attempt back to GraphQL

?

Reply:
That sounds feasible. Resolving the OID and constructing the event body will be the sketchy part.